ts/src/zebpayspot.ts

```diff
@@ -0,0 +1,1204 @@
+        }
+        const endtime = this.safeString (params, 'endTime');
+        const starttime = this.safeString (params, 'startTime');
+        params = this.omit (params, [ 'endtime' ]);
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request: Dict = {
+            'symbol': market['id'],
+        };
+        if (limit === undefined) {
+            limit = 200; // default is 200
+        }
+        request['startTime'] = starttime;
+        request['limit'] = limit; // max 200, default 200
+        request['endTime'] = endtime;
+        request['interval'] = timeframe;
+        let response = undefined;
+        if (market['spot']) {
+            request['category'] = 'spot';
+            response = await this.publicGetMarketKlines (this.extend (request, params));
+        }
+        //
+        //             [
+        //                 [
+        //                     "1670608800000",
+        //                     "17071",
+        //                     "17073",
+        //                     "17027",
+        //                     "17055.5",
+        //                     "268611",
+        //                     "15.74462667"
+        //                 ],
+        //                 [
+        //                     "1670605200000",
+        //                     "17071.5",
+        //                     "17071.5",
+        //                     "17061",
+        //                     "17071",
+        //                     "4177",
+        //                     "0.24469757"
+        //                 ],
+        //                 [
+        //                     "1670601600000",
+        //                     "17086.5",
+        //                     "17088",
+        //                     "16978",
+        //                     "17071.5",
+        //                     "6356",
+        //                     "0.37288112"
+        //                 ]
+        //             ]
+        //
+        return this.parseOHLCVs (response, market, timeframe, since, limit);
+    }
+
+    /**
+     * @method
+     * @name zebpayspot#fetchOrder
+     * @description fetches information on an order made by the user
+     * @see https://github.com/zebpay/zebpay-api-references/blob/main/spot/api-reference/private-endpoints.md#get-order-details
+     * @param {string} id order id
+     * @param {string} symbol unified symbol of the market the order was made in
+     * @param {object} [params] extra parameters specific to the exchange API endpoint
+     * @param {string} [params.timestamp] cancel order by client order id
+     * @returns {object} An [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+     */
+    async fetchOrder (id: Str, symbol: Str = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request: Dict = {};
+        const timestamp = this.safeInteger (params, 'timestamp');
+        if (id === undefined) {
```

Comment:
id is required here we don't need the check,

also again this timestamp is not a unified ccxt term, so we shouldn't use it like that